modules/Movecraft/src/main/java/net/countercraft/movecraft/commands/SiegeCommand.java

```diff
@@ -166,6 +168,30 @@ private boolean beginCommand(CommandSender commandSender){
             player.sendMessage(MOVECRAFT_COMMAND_PREFIX + I18nSupport.getInternationalisedString("The time is not during the Siege schedule"));
             return true;
         }
+
+        //check if piloting craft in siege region
+        Craft siegeCraft = CraftManager.getInstance().getCraftByPlayer(player);
+        if(siegeCraft == null) {
+            player.sendMessage(MOVECRAFT_COMMAND_PREFIX + I18nSupport.getInternationalisedString("You must be piloting a craft!"));
+            return true;
+        }
+        else {
+            if(!siege.getCraftsToWin().contains(siegeCraft.getType().getCraftName())) {
+                player.sendMessage(MOVECRAFT_COMMAND_PREFIX + I18nSupport.getInternationalisedString("You must be piloting a craft that can siege!"));
+                return true;
+            }
+
+            HashHitBox hitBox = siegeCraft.getHitBox();
+            int midX = (hitBox.getMaxX() + hitBox.getMinX()) / 2;
+            int midY = (hitBox.getMaxY() + hitBox.getMinY()) / 2;
+            int midZ = (hitBox.getMaxZ() + hitBox.getMinZ()) / 2;
```

Comment:
use `HitBox::getMidPoint`

Reply:
I was copying from the other siege code, I'll fix both places!